Packlink/Services/BusinessLogic/ShopOrderService.php

```diff
@@ -0,0 +1,360 @@
+<?php
+
+namespace Packlink\Services\BusinessLogic;
+
+use Logeecom\Infrastructure\ORM\QueryFilter\Operators;
+use Logeecom\Infrastructure\ORM\QueryFilter\QueryFilter;
+use Logeecom\Infrastructure\ORM\RepositoryRegistry;
+use Logeecom\Infrastructure\ServiceRegister;
+use Packlink\BusinessLogic\Configuration;
+use Packlink\BusinessLogic\Http\DTO\ParcelInfo;
+use Packlink\BusinessLogic\Http\DTO\Shipment;
+use Packlink\BusinessLogic\Order\Exceptions\OrderNotFound;
+use Packlink\BusinessLogic\Order\Interfaces\ShopOrderService as BaseShopOrderService;
+use Packlink\BusinessLogic\Order\Objects\Address;
+use Packlink\BusinessLogic\Order\Objects\Item;
+use Packlink\BusinessLogic\Order\Objects\Order;
+use Packlink\Entities\OrderDropoffMap;
+use Packlink\Entities\ShippingMethodMap;
+use Shopware\Models\Article\Article;
+
+class ShopOrderService implements BaseShopOrderService
+{
+    /**
+     * @var \Logeecom\Infrastructure\ORM\Interfaces\RepositoryInterface
+     */
+    protected $orderDetailsRepository;
+    /**
+     * @var \Logeecom\Infrastructure\ORM\Interfaces\RepositoryInterface
+     */
+    protected $orderDropoffRepository;
+    /**
+     * @var \Logeecom\Infrastructure\ORM\Interfaces\RepositoryInterface
+     */
+    protected $shippingMethodMapRepository;
+    /**
+     * @var \Packlink\BusinessLogic\Configuration
+     */
+    protected $configService;
+
+    /**
+     * Fetches and returns system order by its unique identifier.
+     *
+     * @param string $orderId $orderId Unique order id.
+     *
+     * @return Order Order object.
+     *
+     * @throws \Logeecom\Infrastructure\ORM\Exceptions\QueryFilterInvalidParamException
+     * @throws \Logeecom\Infrastructure\ORM\Exceptions\RepositoryNotRegisteredException
+     * @throws \Packlink\BusinessLogic\Order\Exceptions\OrderNotFound When order with provided id is not found.
+     */
+    public function getOrderAndShippingData($orderId)
+    {
+        /** @var \Shopware\Models\Order\Order $sourceOrder */
+        $sourceOrder = $this->getShopwareOrderRepository()->find($orderId);
+
+        if ($sourceOrder === null) {
```

Comment:
Do you think we could improve readability catching nulpointerExcetion and thrown exception?

Reply:
No. I think this is the clearest was to do it.

Reply:
In my opinion it is break the success flow with the failed flow . It could use try-catch because it is very simple to read the difference between flows . In this case it is implementing a feature implemented in language , it is implementing behaviour of try-catch but it is implementing in the middle of code.

Reply:
If an order could not be found, we want to stop the execution immediately in a controlled manner.
 
In PHP, accessing a null reference will throw a fatal error that cannot be caught with a `catch` clause ([believe it or not](https://stackoverflow.com/questions/277224/how-do-i-catch-a-php-fatal-error)). The whole point behind our implementation is avoiding this, hence the null condition. In such cases, we want to throw a controlled exception that can be caught and processed.

Reply:
Ok , thanks for explain it